processor/processor.go

```diff
@@ -157,15 +159,20 @@ func (p *Processor) Process(fset *token.FileSet, file ast.File) error {
 			return true
 		}
 
-		fn.Body.List = append(p.Instrumenter.PrefixStatements(spanName, hasError), fn.Body.List...)
-
-		c.Replace(fn)
+		ps := p.Instrumenter.PrefixStatements(spanName, hasError)
+		patches = append(patches, patch{pos: int(fn.Body.Pos()), stmts: ps})
```

Comment:
better to pass dedicated type. it is a bit confusing what this is actually.
```suggestion
		patches = append(patches, patch{pos: fn.Body.Pos(), stmts: ps})
```